includes/DataObjects/Ban.php

```diff
@@ -164,20 +170,14 @@ public function save()
             // update
             $statement = $this->dbObject->prepare(<<<SQL
 UPDATE `ban`
-SET duration = :duration, active = :active, user = :user, action = :action, visibility = :visibility, 
-    targetqueue = :targetqueue, updateversion = updateversion + 1
+SET active = :active, updateversion = updateversion + 1
 WHERE id = :id AND updateversion = :updateversion;
 SQL
             );
             $statement->bindValue(':id', $this->id);
             $statement->bindValue(':updateversion', $this->updateversion);
 
-            $statement->bindValue(':duration', $this->duration);
             $statement->bindValue(':active', $this->active);
-            $statement->bindValue(':user', $this->user);
-            $statement->bindValue(":action", $this->action);
-            $statement->bindValue(":targetqueue", $this->targetqueue);
-            $statement->bindValue(":visibility", $this->visibility);
```

Comment:
#641 asked for the ability to edit bans. Currently bans are kept forever, so the ability to edit a ban would mean losing the ability to see what was banned at a certain time (why a request was rejected etc).

By making this a replace operation under the hood rather than an actual edit, we persist that history.